tool/tsh/common/app_local_proxy.go

```diff
@@ -96,18 +143,30 @@ func (a *localProxyApp) Close() error {
 }
 
 // startLocalALPNProxy starts the local ALPN proxy.
-func (a *localProxyApp) startLocalALPNProxy(ctx context.Context, port string, withTLS bool, opts ...alpnproxy.LocalProxyConfigOpt) error {
+func (a *localProxyApp) startLocalALPNProxy(ctx context.Context, portMapping client.PortMapping, withTLS bool, opts ...alpnproxy.LocalProxyConfigOpt) error {
+	routeToApp := a.appInfo.RouteToApp
+	if portMapping.TargetPort != 0 {
+		routeToApp.TargetPort = uint32(portMapping.TargetPort)
+	}
 	// Create an app cert checker to check and reissue app certs for the local app proxy.
-	appCertChecker := client.NewAppCertChecker(a.tc, a.appInfo.RouteToApp, nil, client.WithTTL(a.tc.KeyTTL))
+	appCertChecker := client.NewAppCertChecker(a.tc, routeToApp, nil, client.WithTTL(a.tc.KeyTTL))
 
 	// If a stored cert is found for the app, try using it.
 	// Otherwise, let the checker reissue one as needed.
-	cert, err := loadAppCertificate(a.tc, a.appInfo.RouteToApp.Name)
+	cert, err := loadAppCertificate(a.tc, routeToApp.Name)
 	if err == nil {
-		appCertChecker.SetCert(cert)
+		if a.app != nil && len(a.app.GetTCPPorts()) > 0 {
+			// There are too many cases to cover when dealing with a multi-port app and an existing cert.
+			// We'd need to consider portMapping passed from argv and TargetPort in the cert.
+			// As tsh proxy app is not the recommended way to access multi-port apps, let's bail out of
+			// using the existing cert and instead always generate a new one.
```

Comment:
Will the new port-specific certificate get stored? If not, and multi-port certs are never supposed to be stored, should we delete the stale on-disk certificate?